connect/src/com/telenor/connect/SdkProfile.java

```diff
@@ -24,10 +24,23 @@
     WellKnownAPI.WellKnownConfig getWellKnownConfig();
     boolean isInitialized();
 
-    void onStartAuthorization(OnStartAuthorizationCallback callback);
+    void initializeFromUri(
+            Map<String, String> parameters,
+            List<String> uiLocales,
+            Uri initFrom,
+            OnDeliverAuthorizationUriCallback callback);
 
-    interface OnStartAuthorizationCallback {
-        void onSuccess();
+    void onStartAuthorization(
+            Map<String, String> parameters,
+            List<String> uiLocales,
+            OnStartAuthorizationCallback callback);
+
+    interface OnDeliverAuthorizationUriCallback {
+        void onSuccess(Uri authorizationStartUri);
         void onError();
     }
+
+    interface OnStartAuthorizationCallback extends OnDeliverAuthorizationUriCallback {
```

Comment:
1. What I meant was that it's developers that publish activities in the
manifest, so I think it's generally detrimental that they have to publish
activities they are not supposed to use themselves, merely to keep the SDK
in shape.
2. I am not sure it's a wrong hook... my understanding is that the hook is
there to answer "yes" if you don't want the url to be rendered by the
WebView, giving the application a chance to do something else instead.